CMakeLists.txt

```diff
@@ -49,7 +49,12 @@ if (BUILD_BACKEND)
 endif()
 
 set(DEPS_BUILD_PATH "${PROJECT_BINARY_DIR}/third_party" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
-set(DEPS_INSTALL_PATH "${DEPS_BUILD_PATH}/install" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
+
+if(NOT DEFINED CMAKE_INSTALL_PREFIX)
+  set(DEPS_INSTALL_PATH "${DEPS_BUILD_PATH}/install" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
+else()
+  set(DEPS_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}/third_party/install" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
+endif()
```

Comment:
Not sure if I am comfortable with this :sweat_smile:. This essentially changes the meaning of `CMAKE_INSTALL_PREFIX`, which is supposed to install MAVSDK (not MAVSDK + a bunch of libraries).

In fact, I would almost be in favor of removing the whole "automated superbuild" logic (i.e. keeping `third_party/`, but not running it at configure time) and documenting how to use it (as a "helper subproject" rather than a "superbuild", if I may say).

So building "with the superbuild" would be something like this:

```cmake
# Build the dependencies and install them locally in third_party/build
cmake -DCMAKE_INSTALL_PREFIX=/path/to/install/dependencies -Bthird_party/build -Sthird_party
cmake --build third_party/build

# Build MAVSDK, using the dependencies above
cmake -DCMAKE_PREFIX_PATH=/path/to/install/dependencies -Bbuild/default -S.
cmake --build build/default
```

In other words, that would make it less automated (and force the user to understand that there is a step where they need to think about the dependencies), and not more automated.

My usual fear with more automation is to slowly but safely move towards the PX4 build system, where every step was surely well-meant, but contributed making it convoluted and very custom.

What do you think?